akka-http-core/src/main/scala/akka/http/ServerSettings.scala

```diff
@@ -46,7 +46,10 @@ final case class ServerSettings(
 
 object ServerSettings extends SettingsCompanion[ServerSettings]("akka.http.server") {
   final case class Timeouts(idleTimeout: Duration,
+                            requestTimeout: Duration,
                             bindTimeout: FiniteDuration) {
+    require(idleTimeout > Duration.Zero, "idleTimeout must be infinite or > 0")
+    require(requestTimeout > Duration.Zero, "requestTimeout must be infinite or > 0")
     require(bindTimeout > Duration.Zero, "bindTimeout must be > 0")
   }
```

Comment:
I remember that this caused some pains when making 2.3/2.4 compatible... I imagine there will be still some movement and additions in http settings so we should perhaps keep it nimble, will check if I remember correctly with the guys on monday.